qa/multi-cluster-search/src/test/java/org/elasticsearch/search/MultiClusterSearchYamlTestSuiteIT.java

```diff
@@ -33,27 +36,36 @@ public static void determineRemoteClusterMinimumVersion() {
         }
     }
 
+    @Override
     protected ClientYamlTestExecutionContext createRestTestExecutionContext(
         ClientYamlTestCandidate clientYamlTestCandidate,
-        ClientYamlTestClient clientYamlTestClient
+        ClientYamlTestClient clientYamlTestClient,
+        final Version esVersion,
+        final Predicate<String> clusterFeaturesPredicate,
+        final String os
     ) {
-        return new ClientYamlTestExecutionContext(clientYamlTestCandidate, clientYamlTestClient, randomizeContentType()) {
+        /*
+         * Since the esVersion is used to skip tests in ESClientYamlSuiteTestCase, we also take into account the
+         * remote cluster version here and return it if it is lower than the local client version. This is used to
+         * skip tests if some feature isn't available on the remote cluster yet.
+         */
+        final Version commonEsVersion;
+        if (remoteEsVersion == null) {
+            commonEsVersion = esVersion;
+        } else {
+            commonEsVersion = remoteEsVersion.before(esVersion) ? remoteEsVersion : esVersion;
+        }
```

Comment:
How about merging the null check into the 2nd statement?
```suggestion
        final Version commonEsVersion = remoteEsVersion != null && remoteEsVersion.before(esVersion) ? remoteEsVersion : esVersion;
```